demo-kotlin/app/src/gms/java/com/example/placesdemo/CurrentPlaceTestActivity.kt

```diff
@@ -60,89 +70,260 @@ class CurrentPlaceTestActivity : AppCompatActivity() {
             Place.Field.OPENING_HOURS,
             Place.Field.PHONE_NUMBER,
             Place.Field.UTC_OFFSET,
-            Place.Field.WEBSITE_URI)
+            Place.Field.WEBSITE_URI
+        )
         fieldSelector = FieldSelector(
             findViewById(R.id.use_custom_fields),
             findViewById(R.id.custom_fields_list),
             savedInstanceState,
-            placeFields)
+            placeFields,
+        )
         responseView = findViewById(R.id.response)
         setLoading(false)
 
         // Set listeners for programmatic Find Current Place
-        findViewById<Button>(R.id.find_current_place_button).setOnClickListener {
-            findCurrentPlace()
-        }
+        findViewById<Button>(R.id.find_current_place_button)
+            .setOnClickListener { findCurrentPlace() }
     }
 
     override fun onSaveInstanceState(bundle: Bundle) {
         super.onSaveInstanceState(bundle)
         fieldSelector.onSaveInstanceState(bundle)
     }
 
+    @SuppressLint("MissingPermission")
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<String>,
+        grantResults: IntArray
+    ) {
+        if (requestCode != PERMISSION_REQUEST_CODE) {
+            super.onRequestPermissionsResult(
+                requestCode,
+                permissions,
+                grantResults
+            )
+            return
+        }
+
+        if (isLocationPermissionGranted(permissions, grantResults)) {
+            findCurrentPlaceWithPermissions()
+        }
+    }
+
+    private fun isLocationPermissionGranted(
+        permissions: Array<String>,
+        grantResults: IntArray
+    ): Boolean =
+        permissions.toList().zip(grantResults.toList())
+            .firstOrNull { (permission, grantResult) ->
+                grantResult == PackageManager.PERMISSION_GRANTED && (ACCESS_FINE_LOCATION == permission || ACCESS_COARSE_LOCATION == permission)
+            } != null
+
     /**
      * Fetches a list of [PlaceLikelihood] instances that represent the Places the user is
-     * most
-     * likely to be at currently.
+     * most likely to be at currently.
      */
+    @SuppressLint("MissingPermission")
     private fun findCurrentPlace() {
-        if (ContextCompat.checkSelfPermission(this, permission.ACCESS_WIFI_STATE)
+        if (ContextCompat.checkSelfPermission(
+                this,
+                ACCESS_FINE_LOCATION
+            )
+            != PackageManager.PERMISSION_GRANTED &&
```

Comment:
Shouldn't this be `||` instead of `&&` for the "or" state?

Reply:
Yep. Good catch. Will update.